client-apis/decrypt.md

```diff
@@ -69,65 +91,226 @@ This default CMM MUST obtain the decryption materials required for decryption.
+this operation MUST decrypt and authenticate the frame (or body) using the
+[authenticated encryption algorithm](../framework/algorithm-suites.md#encryption-algorithm)
+specified by the [algorithm suite](../framework/algorithm-suites.md), with the following inputs:
+
+- The AAD is the serialized [message body AAD](../data-format/message-body-aad.md),
+  constructed as follows:
+  - The [message ID](../data-format/message-body-aad.md#message-id) is the same as the
+    [message ID](../data-frame/message-header.md#message-id) deserialized from the header of this message.
+  - The [Body AAD Content](../data-format/message-body-aad.md#body-aad-content) depends on
+    whether the thing being decrypted is a regular frame, final frame, or unframed data.
+    Refer to [Message Body AAD](../data-format/message-body-aad.md) specification for more information.
+  - The [sequence number](../data-format/message-body-aad.md#sequence-number) is the sequence
+    number deserialized from the frame being decrypted.
+    If this is unframed-data, this value MUST be 1.
+    If this is framed data and the first frame sequentially, this value MUST be 1.
+    Otherwise, this value MUST be 1 greater than the value of the sequence number
+    of the previous frame.
+  - The [content length](../data-format/message-body-aad.md#content-length) MUST have a value
+    equal to the length of the plaintext that was encrypted.
+    This can be determined by using the [frame length](../data-format/message-header.md#frame-length)
+    deserialized from the message header if this is a regular frame,
+    or the [encrypted content length](../data-format/message-body.md#encrypted-content-length)
+    otherwise.
+- The IV is the [sequence number](../data-format/message-body-aad.md#sequence-number)
+  used in the message body AAD above.
+- The cipherkey is the derived data key
+- The ciphertext is the [encrypted content](../data-format/message-body.md#encrypted-content).
+
+If this decryption fails, this operation MUST immediately halt and fail.
+
+If the algorithm suite contains a signature algorithm and
+this operation is streaming the plaintext output to the caller,
+this operation MUST input the serialized frame to the signature algorithm as soon as it is deserialized,
+such that the serialized frame isn't required to remain in memory to complete
+the [signature calculation](#signature-calculation).
+
+If the encrypted message includes an algorithm suite without a signature algorithm,
+the result plaintext MAY be released.
+
+If the encrypted message includes an algorithm suite with a signature algorithm
+the result plaintext MUST NOT be released until [signature verification succeeds](#verify-the-signature-optional).
+
+If streaming the output plaintext for a message with a signature,
+the result plaintext MAY be released, however the caller MUST NOT
+consider this plaintext verified.
+See [security considerations](#security-considerations) below.
```

Comment:
These two paragraphs feel contradictory. Let's clarify it. (I think this comment applies elsewhere.)